src/helpers/parser.ts

```diff
@@ -25,19 +29,52 @@ export const gitIssueParser = async ({ owner, repo, issue_number }: GitParser):
   }
 };
 
-export const gitLinkedIssueParser = async ({ owner, repo, issue_number }: GitParser): Promise<string> => {
+export const gitLinkedIssueParser = async ({ owner, repo, pull_number }: GitParser): Promise<string> => {
   try {
-    const { data } = await axios.get(`https://github.com/${owner}/${repo}/pull/${issue_number}`);
+    const { data } = await axios.get(`https://github.com/${owner}/${repo}/pull/${pull_number}`);
     const dom = parse(data);
     const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
-    const linkedPRs = devForm.querySelectorAll(".my-1");
+    const linkedIssues = devForm.querySelectorAll(".my-1");
 
-    if (linkedPRs.length === 0) {
+    if (linkedIssues.length === 0) {
       return "";
     }
 
-    const prUrl = linkedPRs[0].querySelector("a")?.attrs?.href || "";
-    return prUrl;
+    const issueUrl = linkedIssues[0].querySelector("a")?.attrs?.href || "";
+    return issueUrl;
+  } catch (error) {
+    return "";
+  }
+};
+
+export const gitLinkedPrParser = async ({ owner, repo, issue_number }: GitParser) => {
+  try {
+    const { data } = await axios.get(`https://github.com/${owner}/${repo}/issues/${issue_number}`);
+    const context = getBotContext();
+    const payload = context.payload as Payload;
+    const dom = parse(data);
+    const devForm = dom.querySelector("[data-target='create-branch.developmentForm']") as HTMLElement;
+    const linkedPRs = devForm.querySelectorAll(".my-1");
+    if (linkedPRs.length === 0) return "";
+    let linkedPullRequest;
+    for (const linkedPr of linkedPRs) {
+      const prHref = linkedPr.querySelector("a")?.attrs?.href || "";
+      const parts = prHref.split("/");
+      // extract the organization name and repo name from the link:(e.g. "https://github.com/wannacfuture/ubiquibot/pull/5";)
+      const organization = parts[parts.length - 4];
+      const repository = parts[parts.length - 3];
```

Comment:
Seems a lot more robust to get it from the API based on the current issue instead of the URL.

Reply:
We already get the repo name and org name of the closing issue via API.

But about pull request, we currently have links of the prs. and they can be the prs of another repo, another org.
I don't think it makes sense to get the org name and repo names of the pr using gitHub API because they can be exist on another org's repo.